### ISSUE TITLE
k8s module src directive does not reference files directory when used in a role

### ISSUE THREAD

**Issue by chuckersjp** [ansible-collections/community.kubernetes#70]
##### SUMMARY

The `k8s` module has a `src` directive for specifying the location of files to provide a path to a file containing a valid YAML definition of an object or objects to be created or updated.  It appears that
this requires a full path name.  When used in a role, this directive doesn't appear to use the roles/x/{files,templates,tasks}/ by default.  It will use it if `"{{ role_path }}/files/my_file.yml` is used however.

##### ISSUE TYPE
- Bug Report

##### ANSIBLE VERSION

```
ansible 2.7.8
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Jun 11 2019, 14:33:56) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

##### CONFIGURATION

```[blank]
```

##### OS / ENVIRONMENT

RHEL 7.7

##### STEPS TO REPRODUCE

```
- name: Create secret for htpasswd
  k8s:
    kubeconfig: clusters/{{ cluster }}/ocp4/auth/kubeconfig
    state: present
    src: files/secret.yml
```

##### EXPECTED RESULTS

Secret created based on the contents of `files/secret.yml`

##### ACTUAL RESULTS

Playbook fails with "Error accessing files/secret.yml. Does the file exist?"

```
ansible-playbook 2.7.8
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 2.7.5 (default, Jun 11 2019, 14:33:56) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
Using /etc/ansible/ansible.cfg as config file
/root/ocp4/inventories/hosts.bal1 did not meet host_list requirements, check plugin documentation if this is unexpected
/root/ocp4/inventories/hosts.bal1 did not meet script requirements, check plugin documentation if this is unexpected
/root/ocp4/inventories/hosts.bal1 did not meet yaml requirements, check plugin documentation if this is unexpected
Parsed /root/ocp4/inventories/hosts.bal1 inventory source with ini plugin
 
PLAYBOOK: postconfig.yml ********************************************************************************************
1 plays in postconfig.yml
 
PLAY [localhost] ****************************************************************************************************
META: ran handlers
 
TASK [user-registry-backdoor : Create secret for htpasswd] ******************************************************
task path: /root/ocp4/roles/user-registry-backdoor/tasks/main.yml:1
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: root
<127.0.0.1> EXEC /bin/sh -c 'echo ~root && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1585752693.84-12716652062932 `" && echo ansible-tmp-1585752693.84-12716652062932="` echo /root/.ansible/tmp/ansible-tmp-1585752693.84-12716652062932 `" ) && sleep 0'
Using module file /usr/lib/python2.7/site-packages/ansible/modules/clustering/k8s/k8s.py
<127.0.0.1> PUT /root/.ansible/tmp/ansible-local-48900CduyEt/tmpc6jyoA TO /root/.ansible/tmp/ansible-tmp-1585752693.84-12716652062932/AnsiballZ_k8s.py
<127.0.0.1> EXEC /bin/sh -c 'chmod u+x /root/.ansible/tmp/ansible-tmp-1585752693.84-12716652062932/ /root/.ansible/tmp/ansible-tmp-1585752693.84-12716652062932/AnsiballZ_k8s.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c '/usr/bin/python2 /root/.ansible/tmp/ansible-tmp-1585752693.84-12716652062932/AnsiballZ_k8s.py && sleep 0'
<127.0.0.1> EXEC /bin/sh -c 'rm -f -r /root/.ansible/tmp/ansible-tmp-1585752693.84-12716652062932/ > /dev/null 2>&1 && sleep 0'
fatal: [localhost]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "api_key": null,
            "cert_file": null,
            "context": null,
            "force": false,
            "host": null,
            "key_file": null,
            "kubeconfig": "clusters/bal1/ocp4/auth/kubeconfig",
            "merge_type": null,
            "password": null,
            "ssl_ca_cert": null,
            "state": "present",
            "username": null,
            "verify_ssl": null
        }
    },
    "msg": "Error accessing files/secret.yml. Does the file exist?"
}
        to retry, use: --limit @/root/ocp4/postconfig.retry
 
PLAY RECAP **********************************************************************************************************
localhost                  : ok=0    changed=0    unreachable=0    failed=1
 
```

**Comment by xeor** [comment]
I also got trapped by this.. Most of ansible uses relative path's based on whats in roles.
Including role_path like you did is a very nice solution tho. I'm almost tempted to use that permanent, instead of a "magic" relative lookup..

**Comment by tima** [comment]
@chuckersjp and @xeor -- have you tried the latest Ansible 2.9? Does the issue persist? With moving to collections things are changing a lot. Our ability to back port fixes is also limited at this time as we work through this transition.

**Comment by chuckersjp** [comment]
Had to do some hoop jumping to see if I could replicate this.

Problem still exists for the following:

```
ansible 2.9.10
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Sep 26 2019, 13:23:47) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

**Comment by tima** [comment]
Thanks @chuckersjp. 

@Akasurde could you look into this to see why this is happening? Could you also check roles in collections while you are at it? Thinking like #37 we aren't passing this modules params through all the functions like other modules. Just a thought.

**Comment by Akasurde** [comment]
k8s module can run on the local machine as well as the remote machine (`delegate_to` host) so we might need to implement both `src` and `remote_src` if we go `copy` like `src` functionality.

**Comment by Akasurde** [comment]
This is addressed via #193
